src/test/mod.rs

```diff
@@ -43,10 +47,20 @@ lazy_static! {
         let uri = DEFAULT_URI.clone();
         let mut options = ClientOptions::parse_without_srv_resolution(&uri).unwrap();
         options.max_pool_size = Some(MAX_POOL_SIZE);
+        options.server_api = SERVER_API.clone();
 
         options
     };
     pub(crate) static ref LOCK: TestLock = TestLock::new();
     pub(crate) static ref DEFAULT_URI: String =
         std::env::var("MONGODB_URI").unwrap_or_else(|_| "mongodb://localhost:27017".to_string());
+    pub(crate) static ref SERVER_API: Option<ServerApi> = match std::env::var("MONGODB_API_VERSION")
```

Comment:
From looking below, this is just needed for the server API spec tests, right?

Reply:
This is needed to be able to run the tests against a server with `requireApiVersion=true`. It is used when creating test clients when there's no explicit API version set, but only if the `MONGODB_API_VERSION` environment variable was set.
Without this, the test suite fails all over the place due to the server rejecting commands without `apiVersion` set. For those servers, we run `MONGODB_API_VERSION=1 cargo test --verbose` and our test clients will have an API version declared.

The versioned API spec tests don't need this, as they always declare an API version in the client config.